### PR TITLE
fix: removed unecessary logs for decoding tx in block-explorer

### PR DESCRIPTION
**Pull request by FilipHarald**
## Description

Instead of logging errors for expected behavior it adds 'Unknown ABI' to function name.

Before
Screenshot from 2024-11-15 11-42-04

After
Screenshot from 2024-11-15 11-42-37

## Additional Information

- [x] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [x] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #984_

Your ENS/address: harmont.eth

### CODE REVIEW

File: packages/nextjs/utils/scaffold-eth/decodeTxData.ts [hunk 1]
```diff
@@ -34,12 +35,15 @@ export const decodeTransactionData = (tx: TransactionWithFunction) => {
...
-
+        foundInterface = true;
         break;
       } catch (e) {
-        console.error(`Parsing failed: ${e}`);
+        // NOTE: do nothing
       }
     }
+    if (!foundInterface) {
+      tx.functionName = "Unknown ABI";
```
Comment (technophile-04) [inline packages/nextjs/utils/scaffold-eth/decodeTxData.ts:45]: Shall we do "Unknown Function" here?

**Review by technophile-04 — COMMENTED** [review]
Thanks @FilipHarald for catching and fixing it! 

Do you have an example contract where this happens just to test it once? Lol I am just being lazy but I have encountered a lot of times!

File: packages/nextjs/utils/scaffold-eth/decodeTxData.ts (hunk 1 above)
Comment (FilipHarald) [reply packages/nextjs/utils/scaffold-eth/decodeTxData.ts]: I agree, what do you think about ":warning: Unknown"?

Comment (FilipHarald) [reply packages/nextjs/utils/scaffold-eth/decodeTxData.ts]: Screenshot from 2024-11-20 13-07-53

Comment (technophile-04) [reply packages/nextjs/utils/scaffold-eth/decodeTxData.ts]: Looks good!

**Review by technophile-04 — APPROVED** [review]

**Pull request merged**